src/test/client.test.ts

```diff
@@ -180,3 +180,40 @@ test('should return default variant for disabled toggles', () => {
     expect(result[2].variant?.enabled).toBe(false);
     client.destroy();
 });
+
+test('should register metrics', () => {
+    const config = createProxyConfig({
+        unleashApiToken: '123',
+        unleashUrl: 'http://localhost:4242/api',
+        proxySecrets: ['s1'],
+        environment: 'never-change-me',
+        logLevel: LogLevel.error,
+    });
+
+    const { client, metrics } = createFakeClient(config);
+
+    client.registerMetrics({
+        bucket: {
+            toggles: {
+                toggle: {
+                    yes: 3,
+                    no: 1,
+                    variants: { variantA: 2, variantB: 1, disabled: 1 },
+                },
+            },
+        },
+    });
+
+    expect(metrics.recordedCount).toStrictEqual([
+        ['toggle', true],
+        ['toggle', true],
+        ['toggle', true],
+        ['toggle', false],
+    ]);
+    expect(metrics.recordedCountVariant).toStrictEqual([
+        ['toggle', 'variantA'],
+        ['toggle', 'variantA'],
+        ['toggle', 'variantB'],
+        ['toggle', 'disabled'],
```

Comment:
Is the order significant here? Would it be better to compare counts? Is the storage the same in the FakeMetrics class as in the real Metrics class?

Reply:
FakeMetrics is a very stupid mock recording all invocations. Another alternative would be to have a smarter mock summing up the numbers. I decided to go for the simples fake replacement possible in this case. If you feel strongly about recording the actual numbers I may reconsider my tradeoff.

Reply:
Got it. I don't really feel strongly one way or the other, so no need to change it for me. Just curious 😄